### ISSUE TITLE
pylint is treating numpy.ndarray as if it were a tuple

### ISSUE THREAD

**Issue by mattconway1984** [PyCQA/pylint#2721]
Package versions:

astroid==2.2.0.dev0
pylint==2.3.0.dev1
numpy==1.16.0

### Steps to reproduce

Using the following Python script:

```Python
"""
test.py
A script to reproduce a pylint bug
"""

import numpy as np

A = np.random.randint(9, size=(3, 3))
B = np.zeros_like(A)
A.itemset(4, 0)
B.itemset(4, 0)
```

Execute pylint:

    $ python -m pylint test.py

### Current behavior

The second numpy array (named B) which is formed as a result of calling numpy.zeros_like() causes the following pylint error:

    ************* Module test
    test.py:10:0: E1101: Instance of 'tuple' has no 'itemset' member (no-member)
    
    ------------------------------------------------------------------
    Your code has been rated at 0.00/10 (previous run: 2.86/10, -2.86)

### Expected behavior

No pylint errors, B is not a tuple, it's a perfectly valid numpy.ndarray object.

### pylint --version output
    pylint 2.3.0-dev1
    astroid 2.2.0-dev
    Python 3.6.6 | packaged by conda-forge | (default, Oct 12 2018, 14:08:43) 
    [GCC 4.8.2 20140120 (Red Hat 4.8.2-15)]

**Comment by mattconway1984** [comment]
After some more digging, I discovered that the project I'm working on was previously built a few weeks back, and discovered it was built against numpy version 1.15.4. So I tried rolling back to this issue and re-running my test, the issue did not rear it's head, so I looked a bit deeper:

In numpy version 1.15.4 the **zeros_like** API was simply a global function in numpy.core.numeric (line 94).

Whereas in numpy version 1.16.0 the **zeros_like** API has now been decorated with some decorator named **@array_function_dispatch**. A quick google search brought me to this page:

https://www.numpy.org/neps/nep-0018-array-function-protocol.html

Where I discovered the implementation of this decorator:
```Python
def array_function_dispatch(dispatcher):
    """Wrap a function for dispatch with the __array_function__ protocol."""
    def decorator(implementation):
        @functools.wraps(implementation)
        def public_api(*args, **kwargs):
            relevant_args = dispatcher(*args, **kwargs)
            return implement_array_function(
                implementation, public_api, relevant_args, args, kwargs)
        return public_api
    return decorator
```

Right now I don't have more time to do more digging, but I assume this is probably quite vital information for the issue raised.

**Comment by SimonBiggs** [comment]
I am also having this issue. See Azure build logs:

https://dev.azure.com/pymedphys/pymedphys/_build/results?buildId=286&view=logs&jobId=153be3dc-54a1-5e46-1300-4474530c9a56&taskId=ccd5b9ab-3b80-5665-8915-db3ed3468151&lineStart=671&lineEnd=685&colStart=1&colEnd=130

I have temporarily pinned to numpy 1.15 to rectify this issue within `pymedphys`.

**Comment by hippo91** [comment]
@mattconway1984 thanks for the bug report and for your research. 
I'm working on a fix. Currently i have got one but i'm trying to do a better fix by analyzing the decorator.
However it is a bit complicated.

**Comment by mattconway1984** [comment]
> @mattconway1984 thanks for the bug report and for your research.
> I'm working on a fix. Currently i have got one but i'm trying to do a better fix by analyzing the decorator.
> However it is a bit complicated.

I know, when I first came across the bug and started investigating I thought it could be a quick fix to pylint but then I saw the decorator code from numpy which was tricky to understand and didn't have enough time.... I'm currently ignoring pylint errors raised as a result of this issue but will be nice if it gets fixed properly.

Thanks for looking into it :)

**Comment by raaaaaymond** [comment]
Can confirm the same bug on:

pylint 2.3.1
astroid 2.2.3
Python 3.7.2 (default, Jan 30 2019, 16:09:38) 
[Clang 9.1.0 (clang-902.0.39.2)]
Numpy 1.16.0
